worker/uniter/runner/context/contextfactory.go

```diff
@@ -81,30 +81,41 @@ type contextFactory struct {
 
 	// For generating "unique" context ids.
 	rand *rand.Rand
+
+	slaLevelGetter SLALevelGetter
+}
+
+// SLALevelGetter defines an interface the context factory uses to retrieve
+// the current SLA level.
+type SLALevelGetter interface {
```

Comment:
This abstraction is fine if it is helpful, but if it is on uniterFacade, you probably don't need it.